src/game_data.rs

```diff
@@ -168,11 +180,23 @@ impl<'a, 'b> GameDataBuilder<'a, 'b> {
     ///     // It is legal to register a system with an empty name
     ///     .with(NopSystem, "", &[]);
     /// ~~~
-    pub fn with<S>(mut self, system: S, name: &str, dependencies: &[&str]) -> Self
+    pub fn with<SD, S>(
```

Comment:
I still don't understand how it will be possible to add systems not implementing SystemDesc without having to put them in a bundle. Did I miss something?

Here's how one of my dispatchers looks like, for reference:
![image](https://user-images.githubusercontent.com/4755185/62423045-34dab000-b68a-11e9-9c0a-48ff0675aeb6.png)

In there, there's quite a few systems that are defined externally, in crates that have no dependency on amethyst. This PR would mean that I need to put either ALL systems in a single bundle, or make multiple bundles to add the systems that are not implementing SystemDesc. Did I get that right?

Reply:
> in crates that have no dependency on amethyst

ah, this would make it inconvenient, but not impossible. For the crate that *does* depend on `amethyst` and those systems (i.e. the one that references `GameData`), you can implement `SystemDesc` on a separate struct in that crate.

Alternatives:

* Since the systems appear to be constructible without parameters (I may be wrong), it wouldn't be difficult to write a macro to generate a `SomethingSystemDesc` -> `SomethingSystem`. It doesn't need to be a proc macro.
* Crates that provide systems depend on `amethyst`.
* Crates that provide systems depend on `amethyst_core` and optionally `amethyst_derive` for the proc macro.
* We move the trait to `specs`, though it doesn't really appear to be a `specs` concern (lazy instantiation is only required because Amethyst inserts resources into the `World`).

Reply:
Would like to mention, updating the crates that don't depend on `amethyst` (but do depend on `specs`) would not be such a bad thing -- `amethyst` is compiled for the project anyway, and if they do depend on `amethyst`, the `specs` version is always kept consistent.

The con would be, if you build a crate via `cargo build -p <crate>`, then it would compile `amethyst` separately for that crate instead of `specs` downwards.

Reply:
ha, alternatively, we just expose another `with_` method that takes a `System` directly :smile:
why complicate things

Reply:
Okay, implemented this, `with` takes a `S: System` as usual, `with_system_desc` takes the `SD: SystemDesc`